ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3GetHandler.java

```diff
@@ -0,0 +1,86 @@
+ *
+ */
+package com.github.ambry.frontend.s3;
+
+import com.github.ambry.commons.Callback;
+import com.github.ambry.frontend.AccountAndContainerInjector;
+import com.github.ambry.frontend.FrontendMetrics;
+import com.github.ambry.frontend.GetBlobHandler;
+import com.github.ambry.frontend.SecurityService;
+import com.github.ambry.rest.RequestPath;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestServiceErrorCode;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.router.ReadableStreamChannel;
+
+import static com.github.ambry.rest.RestUtils.InternalKeys.*;
+
+
+/**
+ * Handles S3 Get requests.
+ */
+public class S3GetHandler extends S3BaseHandler<ReadableStreamChannel> {
+  private final SecurityService securityService;
+  private final FrontendMetrics frontendMetrics;
+  private final AccountAndContainerInjector accountAndContainerInjector;
+  private final S3ListHandler s3ListHandler;
+  private final S3MultipartUploadHandler s3MultipartUploadHandler;
+  private final GetBlobHandler getBlobHandler;
+
+  /**
+   * Constructs a handler for uploading s3 GET requests.
+   * @param s3ListHandler S3ListHandler
+   * @param s3MultipartUploadHandler S3MultipartUploadHandler
+   * @param getBlobHandler GetBlobHandler
+   * @param securityService the {@link SecurityService} to use.
+   * @param frontendMetrics {@link FrontendMetrics} instance where metrics should be recorded.
+   * @param accountAndContainerInjector helper to resolve account and container for a given request.
+   */
+  public S3GetHandler(S3ListHandler s3ListHandler, S3MultipartUploadHandler s3MultipartUploadHandler,
+      GetBlobHandler getBlobHandler, SecurityService securityService, FrontendMetrics frontendMetrics,
+      AccountAndContainerInjector accountAndContainerInjector) {
+    this.securityService = securityService;
+    this.frontendMetrics = frontendMetrics;
+    this.accountAndContainerInjector = accountAndContainerInjector;
+    this.s3ListHandler = s3ListHandler;
+    this.s3MultipartUploadHandler = s3MultipartUploadHandler;
+    this.getBlobHandler = getBlobHandler;
+  }
+
+  /**
+   * Handles a S3 Get Request.
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   * @throws RestServiceException exception when the processing fails
+   */
+  @Override
+  protected void doHandle(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      Callback<ReadableStreamChannel> callback) throws RestServiceException {
+    RequestPath requestPath = (RequestPath) restRequest.getArgs().get(REQUEST_PATH);
+    if (S3BaseHandler.isGetObjectLockConfigurationRequest(restRequest)) {
```

Comment:
Nit, you can remove "S3BaseHander."

Reply:
We still need S3BaseHandler, right? We have all the helper functions.
The S3Gethandler is for Get only.